AnkiHubSoftware/ankihub_addon
# feat: Make the Create Copy browser context menu action usable with AnkiHub notes

Closes #437.

Currently Anki's builtin Create Copy context menu action in the browser is not usable with AnkiHub notes because it copies the `ankihub_id` field of the source note to the destination note. The `ankihub_id` field is not visible to the user and should be left unpopulated when using the Create Copy action.

![image](https://user-images.githubusercontent.com/31575114/222903161-56274910-014d-44db-a1c5-058bf1a04dda.png)

This PR fixes this by clearing the `ankihub_id` field in the `AddCards` dialog after the new note is opened in this dialog when the Create Copy action is used.

**How to reproduce**
- Select a note that was downloaded from AnkiHub in the Anki browser
- Use the Create Copy action
- Edit the first field of the note in the Add Cards dialog and save the new note (to make it discernible from the original note)
- Verify that the new note does not have an ankihub id by using the `ID Pending` treeview item and checking that the new note is shown in the search results. (`ID Pending` is the category for AnkiHub notes that don't have an ankihub id.)
<img height=200 src="https://user-images.githubusercontent.com/31575114/222903467-2f71327b-fa33-4999-9785-6b65a1d553e9.png">


Related user report:  https://community.ankihub.net/t/is-it-possible-to-remove-the-ankihub-id-of-a-card/572/2
